src/app/onprem/license-nag-message-management/nag-message.service.js

```diff
@@ -40,25 +41,51 @@ export class NagMessageService {
       return;
     }
     // create nag message element
-    let item = angular.element('<cdvy-nag-message  id="codenvy-nag-message"></cdvy-nag-message>');
+    let item = angular.element('<cdvy-nag-message></cdvy-nag-message>');
+    item.attr('id', this.nagMessageId);
     // compile
     this.nagMessageElement = this.$compile(item)(angular.element(this.$document.find('body')[0]).scope());
   }
 
   /**
    * Show nag message
+   * @returns {boolean} - true if successful
    */
   showLicenseMessage() {
-    this.$document.find('body').addClass('license-message-indent');
-    this.cheBodyInjectorSvc.addElement(this.nagMessageElement);
+    //the parent of the new element
+    let parentElement = this.$document.find('body');
+    let itemId = this.nagMessageElement.attr('id');
+
+    if (!itemId) {
+      return false;
+    }
+
+    let oldItem = this.$document[0].getElementById(itemId);
+    if (oldItem) {
+      oldItem.remove();
+    } else {
+      parentElement.addClass('license-message-indent');
+    }
+
+    parentElement.append(this.nagMessageElement);
+    return true;
   }
 
+
   /**
    * hide nag message
+   * @returns {boolean} - true if successful
    */
   hideLicenseMessage() {
-    this.cheBodyInjectorSvc.removeElement(this.nagMessageElement);
-    this.$document.find('body').removeClass('license-message-indent');
+    let findElement = this.$document[0].getElementById(this.nagMessageId);
+
+    if (findElement) {
+      findElement.remove();
```

Comment:
Because in this case, we compile the widget in the loop:
https://github.com/eclipse/che/blob/66884c7573a6b3829f80799c02c310079e768fc2/dashboard/src/components/body-injector/che-body-injector.service.js#L36